internal/printer/graphviz.go

```diff
@@ -0,0 +1,98 @@
+package printer
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/crossplane/crossplane/internal/k8s"
+	"github.com/emicklei/dot"
+	"github.com/goccy/go-graphviz"
```

Comment:
i wanted to take this PR for a test spin to get a better feel for the functionality, but I'm having trouble building on my Mac M1 Max:

```
❯ make
<omitted...>
22:33:30 [ .. ] go build darwin_arm64
github.com/crossplane/crossplane/cmd/crossplane
package github.com/crossplane/crossplane/cmd/crank
	imports github.com/crossplane/crossplane/internal/printer
	imports github.com/goccy/go-graphviz
	imports github.com/goccy/go-graphviz/cgraph
	imports github.com/goccy/go-graphviz/internal/ccall: build constraints exclude all Go files in /Users/jared/go/pkg/mod/github.com/goccy/go-graphviz@v0.1.1/internal/ccall
22:33:33 [FAIL]
```

Is there additional build or environment config needed to build this? I'm not sure if we're already using cgo, but I'm  hesitant to add that complexity to our build and dependencies if we don't absolutely have to. e.g. https://relistan.com/cgo-when-and-when-not-to-use-it

Reply:
Hm no I didn't set any specific build or environment config for go-graphviz to work. 
But coming back to your point about not using cgo, we can make it work without the go-graphviz library. The graph file is built using the dot library. This library doesn't support saving the graph as a PNG that's why I started using go-graphviz in the first place. But the graph can be rendered like this by the user.`<my-command> | dot -Tpng  > test.png && open test.png` 
https://github.com/emicklei/dot#display-your-graph

Reply:
+1 for offloading the rendering / image generation to an external tool. @phisco mentioned that this is symmetrical with what the golang tools do as well and it would remove complexity from our build/dependencies.

Reply:
Referring to https://go.dev/ref/mod#go-mod-graph

Reply:
Also thinking about it now, it is easier to test if the function is working by comparing the dot output instead of checking a PNG file. Reworked the function now as discussed and removed the graphviz library